Client/mods/deathmatch/logic/luadefs/CLuaPedDefs.cpp

```diff
@@ -2481,16 +2481,25 @@ bool CLuaPedDefs::SetPedExitVehicle(CClientPed* pPed)
     return pPed->ExitVehicle();
 }
 
-bool CLuaPedDefs::killPedTask(CClientPed* ped, taskType taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) noexcept
+bool CLuaPedDefs::killPedTask(CClientPed* ped, taskType taskType, std::uint8_t taskNumber, std::optional<bool> gracefully) 
 {
     switch (taskType)
     {
         case taskType::PRIMARY_TASK:
         {
+            if (taskNumber == TASK_PRIORITY_DEFAULT)
+                throw LuaFunctionError("Killing TASK_PRIORITY_DEFAULT is not allowed");
+
+            if (taskNumber > TASK_PRIORITY_MAX)
```

Comment:
```cpp
if (taskNumber >= TASK_PRIORITY_MAX)
```

Reply:
but there is no slot as TASK_PRIORITY_MAX

Reply:
see https://wiki.multitheftauto.com/wiki/Tasks

Reply:
Exactly, so you want to include this in the return condition

Reply:
Your code is currently equivalent to 
```cpp
if (taskNumber > 5)
```
Since there is no 5 (`TASK_PRIORITY_MAX`) if someone passes 5 then it won't return with "Invalid slot task number"